src/javatests/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommandTest.java

```diff
@@ -90,7 +90,7 @@ public void call_db_user_and_db_password_success() throws IOException, SQLExcept
     assertThat(
         cmd.execute(
             "--db-address",
-            "jdbc:hsqldb:mem:my-animalclinic.example",
+            "jdbc:hsqldb:mem:shiny-brand-new-db",
```

Comment:
I feel that this change cure the symptoms not the disease. 

If the leave the test as it is and remove `call_successWithSchemaFilters`, multiple will fail. I don't know why does it work right now, but error message gives a fair cause – we're trying here to access DB without authentication. If you pass `--db-user sa --dp-password ""` tests will pass (`sa` is default user created in HSQLDB)

I would rather make `--db-user` and `--db-password` required, as in real world DBs have it. Then this test will become obsolete.

Reply:
I tried enabling and disabling different sub-tests and the result supports my theory: that the hsqldb:mem in-memory databases auto-registers the provided username+password combination when first initialized, and persists in the memory until all subtests are existed. Hence if the subtest with username "dbc" is executed first, the hsqldb recognizes "dbc" but not ""; and vice versa.

I do agree that db-user and db-password should be marked required. Making that change in this branch, and will adjust tests accordingly.